### PR TITLE
#issue 1583 doc version dropdown fixed

### LINKED ISSUE #1583: [Site]: Clicking on the doc version dropdown will bring the doc page to the first available doc
### Current State

Let's say we're on a plugin page. Now I want to switch doc version. The doc version dropdown shows up on hover, as expected. However if you accidentally (or out of habit) click on the dropdown, it brings the doc page to the first doc. 

For a demo, please see below gif: 

20230509135256_rec_

### Desired State

The "click" on the doc version dropdown should not have a different behaviour than the hover. 

### Would you like to fix this issue?

None

### PR DESCRIPTION
**Pull request by ShantanuSutar**
Fixes: #1583
@kayx23 
Changes: Now, the "click" on the doc version dropdown does not have a different behavior than the hover.

I have used the concept of css specificity here and removed the pointer-events of the anchor tag inside the div, but allowed the pointer-events inside the unordered lists to function.

**Before:** 

https://github.com/apache/apisix-website/assets/92261216/45ef4e5c-9b62-466e-9759-2704be7d0ff4

**After:** 

https://github.com/apache/apisix-website/assets/92261216/4932bf58-f7e0-40a7-9ab1-54d2b0b40806

### CODE REVIEW

File: doc/src/theme/DocSidebar/styles.module.css [hunk 1]
```diff
@@ -124,8 +125,18 @@
...
 }
 
 .sidebarVersionSwitch div {
   display: inline-block !important;
 }
+
+.sidebarVersionSwitch div:hover{
+  cursor: pointer;
+  color: var(--text-color-primary);
+}
```
Comment (SkyeYoung) [inline doc/src/theme/DocSidebar/styles.module.css]:
```suggestion
}

```

**Comment by SkyeYoung** [comment]
It seems that I have arrived a bit late. 🥲

Apart from my comment above, everything else LGTM. Thank you for your excellent work on this PR. 😸 

Due to some security issues, we have temporarily removed Netlify's automated build bot, so reviewers may need more time to complete the review. We apologize for the delay in merging the changes.

cc @bzp2010 @juzhiyuan @Baoyuantop Pls review this.

**Comment by ShantanuSutar** [comment]
No worries about arriving a bit late! Your input is valuable and appreciated. I'm glad to hear that everything else looks good to you 😸

**Review by Neilblaze — DISMISSED** [review]
LGTM, add the changes mentioned by @SkyeYoung & it should be good to go! 🎉

**Review by SkyeYoung — DISMISSED** [review]

**Review by SkyeYoung — APPROVED** [review]

**Review by guoqqqi — APPROVED** [review]

**Pull request merged**